src/analysis/useless_initializer.d

```diff
@@ -25,49 +31,143 @@ final class UselessInitializerChecker : BaseAnalyzer
+    }
+
+    override void visit(const(Unittest) test)
+    {
+        if (skipTests)
+            return;
+        _inTest = true;
+        test.accept(this);
+        _inTest = false;
+    }
+
+    override void visit(const(StructDeclaration) decl)
+    {
+        if (_inTest)
+            return;
+
+        assert(_inStruct.length > 1);
+
+        const string structName = _inStruct[$-2] ?
+            _structStack.back() ~ "." ~ decl.name.text :
+          decl.name.text;
+
+        _structStack.insert(structName);
+        _structCanBeInit[structName] = false;
+        _atDisabled.insert(false);
+        decl.accept(this);
+        _structStack.removeBack();
+        _atDisabled.removeBack();
+    }
+
+    override void visit(const(Declaration) decl)
+    {
+        _inStruct.insert(decl.structDeclaration !is null);
+        decl.accept(this);
+        if (_inStruct.length > 1 && _inStruct[$-2] && decl.constructor &&
+            ((decl.constructor.parameters && decl.constructor.parameters.parameters.length == 0) ||
+            !decl.constructor.parameters))
+        {
+            _atDisabled[$-1] = decl.attributes
+                .canFind!(a => a.atAttribute !is null && a.atAttribute.identifier.text == "disable");
+        }
+        _inStruct.removeBack();
+    }
+
+    override void visit(const(Constructor) decl)
+    {
+        if (_inStruct.length > 1 && _inStruct[$-2] &&
+            ((decl.parameters && decl.parameters.parameters.length == 0) || !decl.parameters))
+        {
+            const bool canBeInit = !_atDisabled[$-1];
+            _structCanBeInit[_structStack.back()] = canBeInit;
+            if (!canBeInit)
+                _structCanBeInit[_structStack.back()] = !decl.memberFunctionAttributes
+                    .canFind!(a => a.atAttribute !is null && a.atAttribute.identifier.text == "disable");
+        }
+        decl.accept(this);
+    }
+
+    // issue 473, prevent to visit delegates that contain duck type checkers.
+    override void visit(const(TypeofExpression)) {}
+
+    // issue 473, prevent to check expressions in __traits(compiles, ...)
+    override void visit(const(TraitsExpression) e)
+    {
+        if (e.identifier.text == "compiles")
+        {
+            return;
+        }
+        else
+        {
+            e.accept(this);
+        }
     }
 
     override void visit(const(VariableDeclaration) decl)
     {
-        foreach (declarator; decl.declarators)
+        if (!decl.type || !decl.type.type2 ||
+            // initializer has to appear clearly in generated ddoc
+            decl.comment !is null ||
+            // issue 474, manifest constants HAVE to be initialized.
+            decl.storageClasses.canFind!(a => a.token == tok!"enum"))
         {
-            if (!decl.type || !decl.type.type2)
-                continue;
-            if (!declarator.initializer || !declarator.initializer.nonVoidInitializer)
-                continue;
+            return;
```

Comment:
Speaking of tabs, this looks like whitespace here. Should we maybe enable a checker on Travis, s.t. we can't push files which are beginning with tabs?